js/src/forum/components/DiscussionPage.js

```diff
@@ -90,9 +91,9 @@ export default class DiscussionPage extends Page {
 
     return (
       <div className="DiscussionPage">
-        {app.cache.discussionList ? (
+        {app.cache.discussionList.hasDiscussions() ? (
           <div className="DiscussionPage-list" config={this.configPane.bind(this)}>
-            {!$('.App-navigation').is(':visible') ? app.cache.discussionList.render() : ''}
+            {!$('.App-navigation').is(':visible') && <DiscussionList state={app.cache.discussionList} />}
```

Comment:
Interesting little change in capabilities: the component is no longer dynamic. 

@datitisev @askvortsov1 What about having the component class be overridable via prop, with `DiscussionList` being the default? Open for arguments...

Reply:
With _the component is no longer dynamic_ do you mean that other extensions can't replace the class like they could set `app.cache.discussionList` before? Not sure if any ext uses that, but if you want to keep the functionality, i'd be fine with that.

Reply:
Sure, but can we make that an attribute of DiscussionListState?

Reply:
I thought (and was hoping) we're trying to keep states and components clearly separated?

Reply:
Well, for modals and alerts, the class of the component is included in the state, right? Because the state of the application is that there is currently a modal with these attrs and this class?

Reply:
Ah right, a reference to the class should be fine.